scripts/pre_commit_linter.py

```diff
@@ -1478,6 +1478,103 @@ def _check_directive_scope(all_files, parsed_js_files):
     return summary_messages
 
 
+def _check_js_property(all_files, parsed_js_files):
+    """This function ensures that all JS files has exactly
+    one component and the name of the component should match
+    the filename.
+    """
+    print 'Starting js property check'
+    print '----------------------------------------'
+    # Select JS files which need to be checked.
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.js') and not filename.endswith('app.js')]
+    failed = False
+    summary_messages = []
+    property_name = ''
+    for filename in files_to_check:
```

Comment:
The functions `_check_directive_scope()` and `_check_sorted_dependencies` have a lot in common with this function as they all check for JS expressions and the components contained in them. I think combining these functions would lead to confusion. That being said, I have added more comments to describe what the statements are doing. Does it looks OK?

Reply:
I think the _common_ elements of these functions should be combined, especially since now we seem to have three repetitions of similar logic. One rule of programming is: don't repeat yourself (DRY).

As a corollary, you don't need to try and combine parts of the functions that are different. Just abstract out the common logic.